fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintenanceTest.java

```diff
@@ -951,7 +1086,10 @@ private static Stream<Arguments> concurrentParameters() {
         return Stream.concat(Stream.of(false),
                 TestConfigurationUtils.onlyNightly(
                         IntStream.range(0, 3).boxed().flatMap(i -> Stream.of(true, false))))
-                .map(Arguments::of);
+                .flatMap(isSynthetic ->
+                    // there are 500 docs, so -1 is unpartitioned, 100 is partitioned into multiple partitions and 100,000 is partitioned into one partition
+                    Stream.of(-1, 100, 100_000)
```

Comment:
With https://github.com/FoundationDB/fdb-record-layer/pull/3712/files I don't think this is necessarily the case. the relationship to partitions is still correct, but it's 400 docs instead of 500.
Perhaps a bit more brittle than ideal.
Maybe instead, have it take an enum (`UNPARTITIONED`, `FEW_PARTITIONS`, `ONE_PARTITION`) and let `concurrentTestWithinTransaction` determine the high watermark.